server/issue.go

```diff
@@ -1006,3 +1009,99 @@ func (s *Server) postInboxIssueActivity(ctx context.Context, issue *api.Issue, a
 
 	return nil
 }
+
+func (s *Server) getTenantDatabases(ctx context.Context, projectID int, databaseList []*api.Database, databaseName string) ([][]*api.Database, error) {
+	deployConfig, err := s.DeploymentConfigService.FindDeploymentConfig(ctx, &api.DeploymentConfigFind{
+		ProjectID: &projectID,
+	})
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch deployment config for project ID: %v", projectID)).SetInternal(err)
+	}
+	if deployConfig == nil {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Deployment config missing for project ID: %v", projectID)).SetInternal(err)
+	}
+	deploySchedule, err := api.ValidateAndGetDeploymentSchedule(deployConfig.Payload)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, "Failed to get deployment schedule").SetInternal(err)
+	}
+	for _, database := range databaseList {
+		if err := s.composeDatabaseRelationship(ctx, database); err != nil {
+			return nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to compose database relationship for database ID %v", database.ID)).SetInternal(err)
+		}
+	}
+	p, err := getPipelineFromDeploymentSchedule(deploySchedule, databaseName, databaseList)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, "Failed to create deployment pipeline").SetInternal(err)
+	}
+	return p, nil
+}
+
+func (s *Server) getTenantDatabaseSchema(ctx context.Context, instance *api.Instance, projectID int, databaseName string) (string, string, error) {
+	// Find all databases in the project.
+	databaseList, err := s.DatabaseService.FindDatabaseList(ctx, &api.DatabaseFind{
+		ProjectID: &projectID,
+	})
+	if err != nil {
+		return "", "", echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch databases in project ID: %v", projectID)).SetInternal(err)
+	}
+
+	pipeline, err := s.getTenantDatabases(ctx, projectID, databaseList, databaseName)
+	if err != nil {
+		return "", "", err
+	}
+	var existDB *api.Database
+	// We try to use an existing tenant with the same environment and fallback of any first tenant found.
```

Comment:
We try to use a database from an existing tenant in the same environment and fallback to any first tenant found.

Reply:
I also feel we should comment on why we choose the current fallback logic. 

This pickup also seems to be arbitrary. If user wants to bootstrap a new environment, they probably want to specify what's the source environment/instance they want to bootstrap from instead of letting backend to pick?

I feel may be we should just create an empty database if not find matching from the same environment.

Or provide a mechanism for user to specify a source database (probably need to add extra UI to do that when creating db).